pkg/controller/batchrelease/batchrelease_controller.go

```diff
@@ -146,6 +158,13 @@ func (r *BatchReleaseReconciler) Reconcile(ctx context.Context, req ctrl.Request
 
 	klog.Infof("Begin to reconcile BatchRelease(%v/%v), release-phase: %v", release.Namespace, release.Name, release.Status.Phase)
 
+	watched, err := util.AddDynamicWatcher(runtimeController, workloadHandler, WatchedSet, release.Spec.TargetRef.WorkloadRef)
+	if err != nil {
+		return reconcile.Result{}, err
+	} else if watched {
+		WatchedSet.Insert(util.GetGVKFrom(release.Spec.TargetRef.WorkloadRef).String())
```

Comment:
shall we protect WatchedSet against synchronization problems?

Reply:
done